libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -244,6 +244,19 @@ void AP_Scripting::thread(void) {
         }
         num_i2c_devices = 0;
 
+        // clear allocated PWM sources
+        for (uint8_t i=0; i<SCRIPTING_MAX_NUM_PWM_SOURCE; i++) {
+            if (_pwm_source[i] != nullptr) {
+                // Unregister interrupt and delete
+                // Not that this does not check if the interrupt was successfully detached
+                // That state is not available
+                _pwm_source[i]->set_pin(-1, "");
+                delete _pwm_source[i];
+                _pwm_source[i] = nullptr;
+            }
+        }
+        num_pwm_source = 0;
```

Comment:
This is potentially slightly dodgy in that we don't check if the interrupt has been detached. We could just not clear the objects so you would run out faster a few restarts.

Reply:
Not deleting would mean that the script tries to register two objects to the same interrupt pin, so presumably it would fail the second time (I have not tested)